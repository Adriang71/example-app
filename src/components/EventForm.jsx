import React from "react";
import { Controller, ErrorMessage } from "react-hook-form";
import TextField from '@material-ui/core/TextField';
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import { DatePicker } from '@material-ui/pickers';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  wrapper: {
    margin: theme.spacing(5),
  },
  form: {
    '& .MuiTextField-root': {
      margin: theme.spacing(1),
      width: 200,
    },
  },
}));

export default function EventForm({
  control,
  errors,
  handleDateChange,
  handleSubmit,
  onSubmit,
  selectedDate,
  setSelectedDate,
  setValue,
}) {
  const classes = useStyles();

  return (
    <Grid
      container
      direction="row"
      justify="center"
      alignItems="center"
    >
      <Grid item className={classes.wrapper}>
        <form className={classes.form} onSubmit={handleSubmit(onSubmit)}>
          <Controller
            as={<TextField />}
            name="name"
            label="Name"
            control={control}
            defaultValue=""
            error={Boolean(errors.name)}
            helperText={
              <ErrorMessage errors={errors} name="name"/>
            }
          />

          <Controller
            as={<TextField />}
            name="lastName"
            label="Lastname"
            control={control}
            defaultValue=""
            error={Boolean(errors.lastName)}
            helperText={
              <ErrorMessage errors={errors} name="lastName"/>
            }
          />

          <Controller
            as={<TextField />}
            name="email"
            label="Email"
            control={control}
            defaultValue=""
            error={Boolean(errors.email)}
            helperText={
              <ErrorMessage errors={errors} name="email"/>
            }
            />

          <DatePicker
            value={selectedDate}
            format="YYYY/MM/DD"
            label="Date"
            error={Boolean(errors.date)}
            onChange={moment => {
              handleDateChange(moment.format("YYYY/MM/DD"));
              setValue("date", moment.format("YYYY/MM/DD"))
            }}
            helperText={
              <ErrorMessage errors={errors} name="date"/>
            }
          />
          <Grid
            container
            direction="row"
            justify="center"
            alignItems="center"
          >
            <Button type="submit" variant="contained" color="primary">
              Submit
            </Button>
          </Grid>
        </form>
      </Grid>
    </Grid>
  );
}

