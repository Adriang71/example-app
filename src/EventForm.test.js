import React from 'react'
import ReactTestUtils from 'react-dom/test-utils';
import { render, fireEvent, act } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { MuiPickersUtilsProvider } from '@material-ui/pickers';
import MomentUtils from '@date-io/moment';
import fetchMock from 'fetch-mock'
import { Provider } from 'react-redux';
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import EventForm from './EventForm';

const TestProvider = ({store, children}) =>
  <MuiPickersUtilsProvider utils={MomentUtils}>
    <Provider store={store}>
      {children}
    </Provider>
  </MuiPickersUtilsProvider>

const mockStore = configureMockStore([thunk]);


describe("form should", () => {
  let emit;

  beforeAll(() => {
    ({ emit } = window._virtualConsole);
  });

  beforeEach(() => {
    fetchMock.post('*', {response: 200}, { overwriteRoutes: false })
    fetchMock.get('*', {response: 200}, { overwriteRoutes: false })
    window._virtualConsole.emit = jest.fn();
  });

  afterAll(() => {
    window._virtualConsole.emit = emit;
    fetchMock.restore();
  });

  it("display properly", () => {
    const store = mockStore({
        isLoading: false,
        events:[]
    });
    const { asFragment, container } = render(
      <TestProvider store={store}>
        <EventForm />
      </TestProvider>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(container).toMatchSnapshot();
  });

  it("submit", async () => {
    const onSubmit = jest.fn();
    const store = mockStore({
      isLoading: false,
      events:[]
    });
    const { container, queryByText } = render(
      <TestProvider store={store}>
        <EventForm onSubmit={onSubmit} />
      </TestProvider>
    );

    const name = container.querySelector('input[name="name"]')
    const lastName = container.querySelector('input[name="lastName"]')
    const email = container.querySelector('input[name="email"]')
    const date = container.querySelector('input[name="date"]')
    const submit = container.querySelector('button[type="submit"]')

    await act(async () => {

      await fireEvent.change(name, {
        target: {
          value: 'mockname'
        }
      })

      await fireEvent.change(lastName, {
        target: {
          value: 'mocklastname'
        }
      })

      await fireEvent.change(email, {
        target: {
          value: 'example@o2.pl'
        }
      })

      await fireEvent.change(date, {
        target: {
          value: '22/02/2022'
        }
      })

      await fireEvent.click(submit);

      expect(onSubmit).toHaveBeenCalled();
      expect(queryByText('Required*')).toBeNull()
    });
  });

});