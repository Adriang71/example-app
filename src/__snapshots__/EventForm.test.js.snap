// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form should display properly 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root makeStyles-wrapper-1 MuiGrid-item"
    >
      <form
        class="makeStyles-form-2"
      >
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
          >
            Name
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              name="name"
              type="text"
              value=""
            />
          </div>
          <p
            class="MuiFormHelperText-root"
          />
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
          >
            Lastname
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              name="lastName"
              type="text"
              value=""
            />
          </div>
          <p
            class="MuiFormHelperText-root"
          />
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
          >
            Email
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              name="email"
              type="text"
              value=""
            />
          </div>
          <p
            class="MuiFormHelperText-root"
          />
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root"
          format="YYYY/MM/DD"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
          >
            Date
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              name="date"
              readonly=""
              type="text"
              value=""
            />
          </div>
          <p
            class="MuiFormHelperText-root"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Submit
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;

exports[`form should display properly 2`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      class="MuiGrid-root makeStyles-wrapper-1 MuiGrid-item"
    >
      <form
        class="makeStyles-form-2"
      >
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
          >
            Name
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              name="name"
              type="text"
              value=""
            />
          </div>
          <p
            class="MuiFormHelperText-root"
          />
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
          >
            Lastname
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              name="lastName"
              type="text"
              value=""
            />
          </div>
          <p
            class="MuiFormHelperText-root"
          />
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
          >
            Email
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              name="email"
              type="text"
              value=""
            />
          </div>
          <p
            class="MuiFormHelperText-root"
          />
        </div>
        <div
          class="MuiFormControl-root MuiTextField-root"
          format="YYYY/MM/DD"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
          >
            Date
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiInput-input"
              name="date"
              readonly=""
              type="text"
              value=""
            />
          </div>
          <p
            class="MuiFormHelperText-root"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              Submit
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
